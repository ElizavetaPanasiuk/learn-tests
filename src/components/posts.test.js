import Posts from './Posts';
import { render, configure, shallow, mount } from 'enzyme';
import toJson from 'enzyme-to-json';
import Adapter from '@wojtekmaj/enzyme-adapter-react-17';

configure({
  adapter: new Adapter()
});

describe('<Posts />', () => {
  let component = shallow(<Posts />);
  let wrapper;

  it('should render include a link', () => {
    wrapper = component.find('.newsList');
    expect(wrapper.length).toBeGreaterThanOrEqual(1);
  });

  it('renders posts', () => {
    expect(toJson(component)).toMatchSnapshot(); // toJSON must be used in new version
  });

  it('renders all nested tags', () => {
    component = render(<Posts />);
    expect(toJson(component)).toMatchSnapshot();
  });
});