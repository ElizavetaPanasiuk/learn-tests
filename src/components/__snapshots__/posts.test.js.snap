// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Posts /> renders all nested tags 1`] = `
<ul
  class="newsList"
>
  <li
    class="post"
  >
    <div
      class="description"
    >
      <a
        class="title"
        href="//test.url"
      >
        Jest & Enzyme
      </a>
      <span
        class="text"
      >
        100 points
      </span>
      <span
        class="comments"
      >
        10 comments
      </span>
      <span
        class="date"
      >
        5/4/2020
      </span>
      <span
        class="author"
      >
        Yauhen
      </span>
    </div>
  </li>
  <li
    class="post"
  >
    <div
      class="description"
    >
      <a
        class="title"
        href="//test2121.url"
      >
        TypeScript Basics
      </a>
      <span
        class="text"
      >
        10 points
      </span>
      <span
        class="comments"
      >
        8 comments
      </span>
      <span
        class="date"
      >
        5/6/2020
      </span>
      <span
        class="author"
      >
        Stepan
      </span>
    </div>
  </li>
</ul>
`;

exports[`<Posts /> renders posts 1`] = `
<ul
  className="newsList"
>
  <Post
    author="Yauhen"
    created_at="2020-05-03T23:36:09.816Z"
    key="1"
    num_comments={10}
    points={100}
    title="Jest & Enzyme"
    url="//test.url"
  />
  <Post
    author="Stepan"
    created_at="2020-05-05T23:36:09.816Z"
    key="2"
    num_comments={8}
    points={10}
    title="TypeScript Basics"
    url="//test2121.url"
  />
</ul>
`;
